auth/lib/ui/home_page.dart

```diff
@@ -110,13 +113,24 @@ class _HomePageState extends State<HomePage> {
       _allCodes = codes;
       hasTrashedCodes = false;
       hasNonTrashedCodes = false;
+      hasNonFavouriteCodes = false;
+      hasFavouriteCodes = false;
+
       for (final c in _allCodes ?? []) {
         if (c.isTrashed) {
           hasTrashedCodes = true;
         } else {
           hasNonTrashedCodes = true;
         }
-        if (hasNonTrashedCodes && hasTrashedCodes) {
+        if (c.isPinned) {
+          hasFavouriteCodes = true;
+        } else {
+          hasNonFavouriteCodes = true;
+        }
```

Comment:
This check we should do for non-trashed codes only. If only pinCodes are in trash, we don't want to show the tab.